src/sharing-client.ts

```diff
@@ -1,43 +1,80 @@
-import {InitMessageName, Context} from "./init-message";
+import {
+  InitMessageName,
+  InitResponseMessage,
+  InitResponseMessageName,
+  Context} from "./init-message";
 import {
   LaunchApplication,
-  Publishable,
+  PublishResponse,
   PublishMessageName,
   PublishResponseMessageName,
   Representation} from "./publishable";
-import { IFramePhone } from "./iframe-phone";
+
+import { IFramePhoneUp, IFramePhoneFactory } from "./iframe-phone";
+
 
 export interface SharableApp {
   application: LaunchApplication;
   getDataFunc(context:Context): Promise<Representation[]>;
 }
 
 export class SharingClient {
-  phone: IFramePhone;
+  phone: IFramePhoneUp;
   context: Context;
   app: SharableApp;
 
-  constructor(phone:IFramePhone, app:SharableApp) {
-    this.phone = phone;
+  constructor(phone:IFramePhoneUp|null, app:SharableApp) {
```

Comment:
Why not reverse the parameters and make phone optional like `phone?:IFramePhoneUp`?